server/src/main/java/io/druid/segment/realtime/appenderator/AppenderatorImpl.java

```diff
@@ -0,0 +1,1192 @@
+  }
+
+  private void resetNextFlush()
+  {
+    nextFlush = new DateTime().plus(tuningConfig.getIntermediatePersistPeriod()).getMillis();
+  }
+
+  /**
+   * Populate "sinks" and "sinkTimeline" with committed segments, and announce them with the segmentAnnouncer.
+   *
+   * @return persisted commit metadata
+   */
+  private Object bootstrapSinksFromDisk()
+  {
+    Preconditions.checkState(sinks.isEmpty(), "Already bootstrapped?!");
+
+    final File baseDir = tuningConfig.getBasePersistDirectory();
+    if (!baseDir.exists()) {
+      return null;
+    }
+
+    final File[] files = baseDir.listFiles();
+    if (files == null) {
+      return null;
+    }
+
+    final File commitFile = computeCommitFile();
+    final Committed committed;
+    try {
+      if (commitFile.exists()) {
+        committed = objectMapper.readValue(commitFile, Committed.class);
+      } else {
+        committed = Committed.nil();
+      }
+    }
+    catch (Exception e) {
+      throw new ISE(e, "Failed to read commitFile: %s", commitFile);
+    }
+
+    log.info("Loading sinks from[%s]: %s", baseDir, committed.getHydrants().keySet());
+
+    for (File sinkDir : files) {
+      final File identifierFile = new File(sinkDir, IDENTIFIER_FILE_NAME);
+      if (!identifierFile.isFile()) {
+        // No identifier in this sinkDir; it must not actually be a sink directory. Skip it.
+        continue;
+      }
+
+      try {
+        final SegmentIdentifier identifier = objectMapper.readValue(
+            new File(sinkDir, "identifier.json"),
+            SegmentIdentifier.class
+        );
+
+        final int committedHydrants = committed.getCommittedHydrants(identifier.getIdentifierAsString());
+
+        if (committedHydrants <= 0) {
+          log.info("Removing uncommitted sink at [%s]", sinkDir);
+          FileUtils.deleteDirectory(sinkDir);
+          continue;
+        }
+
+        // To avoid reading and listing of "merged" dir and other special files
+        final File[] sinkFiles = sinkDir.listFiles(
+            new FilenameFilter()
+            {
+              @Override
+              public boolean accept(File dir, String fileName)
+              {
+                return !(Ints.tryParse(fileName) == null);
+              }
+            }
+        );
+
+        Arrays.sort(
+            sinkFiles,
+            new Comparator<File>()
+            {
+              @Override
+              public int compare(File o1, File o2)
+              {
+                return Ints.compare(Integer.parseInt(o1.getName()), Integer.parseInt(o2.getName()));
+              }
+            }
+        );
+
+        List<FireHydrant> hydrants = Lists.newArrayList();
+        for (File hydrantDir : sinkFiles) {
+          final int hydrantNumber = Integer.parseInt(hydrantDir.getName());
+
+          if (hydrantNumber >= committedHydrants) {
+            log.info("Removing uncommitted segment at [%s]", hydrantDir);
+            FileUtils.deleteDirectory(hydrantDir);
```

Comment:
don't really want the alert, or the log message, or really anything it does other than `FileUtils.deleteDirectory`.